pandas/core/internals.py

```diff
@@ -216,20 +216,25 @@ def make_block(self, values, placement=None, ndim=None):
         if ndim is None:
             ndim = self.ndim
 
-        return make_block(values, placement=placement, ndim=ndim)
+        return make_block(values, placement=placement, ndim=ndim, dtype=dtype)
 
     def make_block_scalar(self, values):
         """
         Create a ScalarBlock
         """
         return ScalarBlock(values)
 
-    def make_block_same_class(self, values, placement=None, ndim=None):
+    def make_block_same_class(self, values, placement=None, ndim=None,
+                              dtype=None):
         """ Wrap given values in a block of same type as self. """
+        if dtype is not None:
+            # issue 19431 fastparquet is passing this
+            warnings.warn("dtype argument is deprecated, will be removed "
+                          "in a future release.", FutureWarning)
```

Comment:
Let's make this a DeprecationWarning instead of FutureWarning, as it is typically only something developers need to see, not users.

Reply:
post an issue on fastparquet to fix this as well

Reply:
let’s leave this as a FutureWarning
it will encourage fp to fix this as it will be visible

Reply:
We can use the issue I opened yesterday: https://github.com/dask/fastparquet/issues/297 

FutureWarning will just be annoying for users in this case, and I am confident fastparquet will change that

Reply:
if it’s changed then we can simply move the bar on fp min version and this is no problem

Reply:
we are not keeping private API around for external packages
this is way too much work

Reply:
gents, which is it? FutureWarning or DeprecationWarning? it seems to me that DeprecationWarning is meant to be used here.

Reply:
pls leave it as FutureWarning

Reply:
ok